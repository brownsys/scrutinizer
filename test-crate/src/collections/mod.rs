mod btreemap;
mod hashmap;
mod hashset;
mod linked_list;
mod vec;
