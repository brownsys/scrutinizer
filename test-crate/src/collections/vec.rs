use crate::redefine;
use std::alloc::Global;
use std::cmp::Ordering;
use std::collections::TryReserveError;
use std::mem::MaybeUninit;
use std::ops::{Range, RangeBounds};
use std::slice::{
    Chunks, ChunksExact, ChunksExactMut, ChunksMut, EscapeAscii, Iter, IterMut, RChunks,
    RChunksExact, RChunksExactMut, RChunksMut, Windows,
};
use std::vec::Drain;

// Natively available methods.
redefine! { <Vec<usize>>::append => vec: &mut Vec<usize>, other: &mut Vec<usize, Global> => (), "impure" }
redefine! { <Vec<usize>>::as_mut_ptr => vec: &mut Vec<usize> => *mut usize, "impure" }
redefine! { <Vec<usize>>::as_mut_slice => vec: &mut Vec<usize> => &mut [usize], "impure" }
redefine! { <Vec<usize>>::as_ptr => vec: &Vec<usize> => *const usize, "pure" }
redefine! { <Vec<usize>>::as_slice => vec: &Vec<usize> => &[usize], "pure" }
redefine! { <Vec<usize>>::capacity => vec: &Vec<usize> => usize, "pure" }
redefine! { <Vec<usize>>::clear => vec: &mut Vec<usize> => (), "impure" }
redefine! { <Vec<usize>>::dedup => vec: &mut Vec<usize> => (), "impure" }
redefine! { <Vec<usize>>::dedup_by => vec: &mut Vec<usize>, same_bucket: impl FnMut(&mut usize, &mut usize) -> bool => (), "impure" }
redefine! { <Vec<usize>>::dedup_by_key => vec: &mut Vec<usize>, key: impl FnMut(&mut usize) -> usize => (), "impure" }
redefine! { <Vec<usize>>::drain => vec: &mut Vec<usize>, range: impl RangeBounds<usize> => Drain<'_, usize, Global>, "impure" }
redefine! { <Vec<usize>>::extend_from_slice => vec: &mut Vec<usize>, other: &[usize] => (), "impure" }
redefine! { <Vec<usize>>::extend_from_within => vec: &mut Vec<usize>, src: impl RangeBounds<usize> => (), "impure" }
redefine! { <Vec<usize>>::insert => vec: &mut Vec<usize>, index: usize, element: usize => (), "impure" }
redefine! { <Vec<usize>>::into_boxed_slice => vec: Vec<usize> => Box<[usize], Global>, "pure" }
redefine! { <Vec<usize>>::is_empty => vec: &Vec<usize> => bool, "pure" }
redefine! { <Vec<usize>>::leak => vec: Vec<usize> => &'static mut [usize], "pure" }
redefine! { <Vec<usize>>::len => vec: &Vec<usize> => usize, "pure" }
redefine! { <Vec<usize>>::new => => Vec<usize, Global>, "pure" }
redefine! { <Vec<usize>>::pop => vec: &mut Vec<usize> => Option<usize>, "impure" }
redefine! { <Vec<usize>>::push => vec: &mut Vec<usize>, value: usize => (), "impure" }
redefine! { <Vec<usize>>::remove => vec: &mut Vec<usize>, index: usize => usize, "impure" }
redefine! { <Vec<usize>>::reserve => vec: &mut Vec<usize>, additional: usize => (), "impure" }
redefine! { <Vec<usize>>::reserve_exact => vec: &mut Vec<usize>, additional: usize => (), "impure" }
redefine! { <Vec<usize>>::resize => vec: &mut Vec<usize>, new_len: usize, value: usize => (), "impure" }
redefine! { <Vec<usize>>::resize_with => vec: &mut Vec<usize>, new_len: usize, f: impl FnMut() -> usize => (), "impure" }
redefine! { <Vec<usize>>::retain => vec: &mut Vec<usize>, f: impl FnMut(&usize) -> bool => (), "impure" }
redefine! { <Vec<usize>>::retain_mut => vec: &mut Vec<usize>, f: impl FnMut(&mut usize) -> bool => (), "impure" }
redefine! { <Vec<usize>>::shrink_to => vec: &mut Vec<usize>, min_capacity: usize => (), "impure" }
redefine! { <Vec<usize>>::shrink_to_fit => vec: &mut Vec<usize> => (), "impure" }
redefine! { <Vec<usize>>::spare_capacity_mut => vec: &mut Vec<usize> => &mut [MaybeUninit<usize>], "impure" }
redefine! { <Vec<usize>>::split_off => vec: &mut Vec<usize>, at: usize => Vec<usize, Global>, "impure" }
redefine! { <Vec<usize>>::swap_remove => vec: &mut Vec<usize>, index: usize => usize, "impure" }
redefine! { <Vec<usize>>::truncate => vec: &mut Vec<usize>, len: usize => (), "impure" }
redefine! { <Vec<usize>>::try_reserve => vec: &mut Vec<usize>, additional: usize => Result<(), TryReserveError>, "impure" }
redefine! { <Vec<usize>>::try_reserve_exact => vec: &mut Vec<usize>, additional: usize => Result<(), TryReserveError>, "impure" }
redefine! { <Vec<usize>>::with_capacity => capacity: usize => Vec<usize, Global>, "pure" }

// Methods implicitly implemented by Deref<Target=[T]>.
redefine! { <[_]>::as_mut_ptr_range => vec: &mut Vec<usize> => Range<*mut usize>, "impure" }
redefine! { <[_]>::as_ptr_range => vec: &Vec<usize> => Range<*const usize>, "pure" }
redefine! { <[_]>::binary_search => vec: &Vec<usize>, x: &usize => Result<usize, usize>, "pure" }
redefine! { <[_]>::binary_search_by<'a> => vec: &'a Vec<usize>, f: impl FnMut(&'a usize) -> Ordering => Result<usize, usize>, "impure" }
redefine! { <[_]>::binary_search_by_key<'a> => vec: &'a Vec<usize>, b: &usize, f: impl FnMut(&'a usize) -> usize => Result<usize, usize>, "impure" }
redefine! { <[_]>::chunks => vec: &Vec<usize>, chunk_size: usize => Chunks<'_, usize>, "pure" }
redefine! { <[_]>::chunks_exact => vec: &Vec<usize>, chunk_size: usize => ChunksExact<'_, usize>, "pure" }
redefine! { <[_]>::chunks_exact_mut => vec: &mut Vec<usize>, chunk_size: usize => ChunksExactMut<'_, usize>, "impure" }
redefine! { <[_]>::chunks_mut => vec: &mut Vec<usize>, chunk_size: usize => ChunksMut<'_, usize>, "impure" }
redefine! { <[_]>::clone_from_slice => vec: &mut Vec<usize>, src: &[usize] => (), "impure" }
redefine! { <[_]>::contains => vec: &Vec<usize>, x: &usize => bool, "pure" }
redefine! { <[_]>::copy_from_slice => vec: &mut Vec<usize>, src: &[usize] => (), "impure" }
redefine! { <[_]>::copy_within => vec: &mut Vec<usize>, src: impl RangeBounds<usize>, dest: usize => (), "impure" }
redefine! { <[_]>::ends_with => vec: &Vec<usize>, needle: &[usize] => bool, "pure" }
redefine! { <[_]>::fill => vec: &mut Vec<usize>, value: usize => (), "impure" }
redefine! { <[_]>::fill_with => vec: &mut Vec<usize>, f: impl FnMut() -> usize => (), "impure" }
redefine! { <[_]>::first => vec: &Vec<usize> => Option<&usize>, "pure" }
redefine! { <[_]>::first_mut => vec: &mut Vec<usize> => Option<&mut usize>, "impure" }
redefine! { <[_]>::get => vec: &Vec<usize>, index: usize => Option<&usize>, "pure" }
redefine! { <[_]>::get_mut => vec: &mut Vec<usize>, index: usize => Option<&mut usize>, "impure" }
redefine! { <[_]>::iter => vec: &Vec<usize> => Iter<'_, usize>, "pure" }
redefine! { <[_]>::iter_mut => vec: &mut Vec<usize> => IterMut<'_, usize>, "impure" }
redefine! { <[_]>::last => vec: &Vec<usize> => Option<&usize>, "pure" }
redefine! { <[_]>::last_mut => vec: &mut Vec<usize> => Option<&mut usize>, "impure" }
redefine! { <[_]>::partition_point => vec: &Vec<usize>, pred: impl FnMut(&usize) -> bool => usize, "impure" }
redefine! { <[_]>::rchunks => vec: &Vec<usize>, chunk_size: usize => RChunks<'_, usize>, "pure" }
redefine! { <[_]>::rchunks_exact => vec: &Vec<usize>, chunk_size: usize => RChunksExact<'_, usize>, "pure" }
redefine! { <[_]>::rchunks_exact_mut => vec: &mut Vec<usize>, chunk_size: usize => RChunksExactMut<'_, usize>, "impure" }
redefine! { <[_]>::rchunks_mut => vec: &mut Vec<usize>, chunk_size: usize => RChunksMut<'_, usize>, "impure" }
redefine! { <[_]>::repeat => vec: &Vec<usize>, n: usize => Vec<usize, Global>, "pure" }
redefine! { <[_]>::reverse => vec: &mut Vec<usize> => (), "impure" }
redefine! { <[_]>::rotate_left => vec: &mut Vec<usize>, mid: usize => (), "impure" }
redefine! { <[_]>::rotate_right => vec: &mut Vec<usize>, k: usize => (), "impure" }
redefine! { <[_]>::select_nth_unstable => vec: &mut Vec<usize>, index: usize => (&mut [usize], &mut usize, &mut [usize]), "impure" }
redefine! { <[_]>::select_nth_unstable_by => vec: &mut Vec<usize>, index: usize, compare: impl FnMut(&usize, &usize) -> Ordering => (&mut [usize], &mut usize, &mut [usize]), "impure" }
redefine! { <[_]>::select_nth_unstable_by_key => vec: &mut Vec<usize>, index: usize, f: impl FnMut(&usize) -> usize => (&mut [usize], &mut usize, &mut [usize]), "impure" }
redefine! { <[_]>::sort => vec: &mut Vec<usize> => (), "impure" }
redefine! { <[_]>::sort_by => vec: &mut Vec<usize>, compare: impl FnMut(&usize, &usize) -> Ordering => (), "impure" }
redefine! { <[_]>::sort_by_cached_key => vec: &mut Vec<usize>, f: impl FnMut(&usize) -> usize => (), "impure" }
redefine! { <[_]>::sort_by_key => vec: &mut Vec<usize>, f: impl FnMut(&usize) -> usize => (), "impure" }
redefine! { <[_]>::sort_unstable => vec: &mut Vec<usize> => (), "impure" }
redefine! { <[_]>::sort_unstable_by => vec: &mut Vec<usize>, compare: impl FnMut(&usize, &usize) -> Ordering => (), "impure" }
redefine! { <[_]>::sort_unstable_by_key => vec: &mut Vec<usize>, f: impl FnMut(&usize) -> usize => (), "impure" }
redefine! { <[_]>::split_at => vec: &Vec<usize>, mid: usize => (&[usize], &[usize]), "pure" }
redefine! { <[_]>::split_at_mut => vec: &mut Vec<usize>, mid: usize => (&mut [usize], &mut [usize]), "impure" }
redefine! { <[_]>::split_first => vec: &Vec<usize> => Option<(&usize, &[usize])>, "pure" }
redefine! { <[_]>::split_first_mut => vec: &mut Vec<usize> => Option<(&mut usize, &mut [usize])>, "impure" }
redefine! { <[_]>::split_last => vec: &Vec<usize> => Option<(&usize, &[usize])>, "pure" }
redefine! { <[_]>::split_last_mut => vec: &mut Vec<usize> => Option<(&mut usize, &mut [usize])>, "impure" }
redefine! { <[_]>::starts_with => vec: &Vec<usize>, needle: &[usize] => bool, "pure" }
redefine! { <[_]>::swap => vec: &mut Vec<usize>, a: usize, b: usize => (), "impure" }
redefine! { <[_]>::swap_with_slice => vec: &mut Vec<usize>, other: &mut [usize] => (), "impure" }
redefine! { <[_]>::to_vec => vec: &Vec<usize> => Vec<usize, Global>, "pure" }
redefine! { <[_]>::windows => vec: &Vec<usize>, size: usize => Windows<'_, usize>, "pure" }

// ASCII-related methods.
redefine! { <[_]>::eq_ignore_ascii_case => vec: &Vec<u8>, other: &[u8] => bool, "pure" }
redefine! { <[_]>::escape_ascii => vec: &Vec<u8> => EscapeAscii<'_>, "pure" }
redefine! { <[_]>::is_ascii => vec: &Vec<u8> => bool, "pure" }
redefine! { <[_]>::make_ascii_lowercase => vec: &mut Vec<u8> => (), "impure" }
redefine! { <[_]>::make_ascii_uppercase => vec: &mut Vec<u8> => (), "impure" }
redefine! { <[_]>::to_ascii_lowercase => vec: &Vec<u8> => Vec<u8, Global>, "pure" }
redefine! { <[_]>::to_ascii_uppercase => vec: &Vec<u8> => Vec<u8, Global>, "pure" }
