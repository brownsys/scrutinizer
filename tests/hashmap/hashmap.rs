use crate::redefine;
use std::collections::hash_map::*;
use std::collections::HashMap;
use std::collections::TryReserveError;
use std::fmt::Debug;
use std::fmt::{Formatter, Result as FmtResult};
use std::ops::Index;

redefine! { <HashMap<usize, usize>>::capacity => hashmap: &HashMap<usize, usize> => usize, "pure" }
redefine! { <HashMap<usize, usize>>::clone => hashmap: &HashMap<usize, usize> => HashMap<usize, usize>, "pure" }
redefine! { <HashMap<usize, usize>>::clone_from => hashmap: &mut HashMap<usize, usize>, other: &HashMap<usize, usize> => (), "impure" }
redefine! { <HashMap<usize, usize>>::clear => hashmap: &mut HashMap<usize, usize> => (), "impure" }
redefine! { <HashMap<usize, usize>>::contains_key => hashmap: &HashMap<usize, usize>, element: &usize => bool, "pure" }
redefine! { <HashMap<usize, usize>>::drain => hashmap: &mut HashMap<usize, usize> => Drain<'_, usize, usize>, "impure" }
redefine! { <HashMap<usize, usize>>::default => => HashMap<usize, usize>, "pure" }
redefine! { <HashMap<usize, usize>>::entry => hashmap: &mut HashMap<usize, usize>, key: usize => Entry<'_, usize, usize>, "impure" }
redefine! { extend_1, <HashMap<usize, usize>>::extend<'a> => hashmap: &mut HashMap<usize, usize>, iter: impl IntoIterator<Item = (&'a usize, &'a usize)> => (), "impure" }
redefine! { extend_2, <HashMap<usize, usize>>::extend => hashmap: &mut HashMap<usize, usize>, iter: impl IntoIterator<Item = (usize, usize)> => (), "impure" }
redefine! { <HashMap<usize, usize>>::eq => hashmap: &HashMap<usize, usize>, other: &HashMap<usize, usize> => bool, "pure" }
redefine! { <HashMap<usize, usize>>::from => arr: [(usize, usize); 10] => HashMap<usize, usize>, "pure" }
redefine! { <HashMap<usize, usize>>::from_iter => iter: impl IntoIterator<Item = (usize, usize)> => HashMap<usize, usize>, "impure" }
redefine! { <HashMap<usize, usize>>::fmt => hashmap: &HashMap<usize, usize>, f: &mut Formatter<'_> => FmtResult, "impure" }
redefine! { <HashMap<usize, usize>>::get<'a> => hashmap: &'a HashMap<usize, usize>, k: &usize => Option<&'a usize>, "pure" }
redefine! { <HashMap<usize, usize>>::get_key_value<'a, 'b> => hashmap: &'a HashMap<usize, usize>, k: &'a usize => Option<(&'a usize, &'a usize)>, "pure" }
redefine! { <HashMap<usize, usize>>::get_mut<'a> => hashmap: &'a mut HashMap<usize, usize>, k: &usize => Option<&'a mut usize>, "impure" }
redefine! { <HashMap<usize, usize>>::hasher => hashmap: &HashMap<usize, usize> => &RandomState, "pure" }
redefine! { <HashMap<usize, usize>>::index<'a> => hashmap: &'a HashMap<usize, usize>, key: &'a usize => &'a usize, "pure" }
redefine! { <HashMap<usize, usize>>::insert => hashmap: &mut HashMap<usize, usize>, k: usize, v: usize => Option<usize>, "impure" }
redefine! { into_iter_1, <HashMap<usize, usize>>::into_iter => hashmap: HashMap<usize, usize> => IntoIter<usize, usize>, "pure" }
#[doc = "impure"]
pub fn into_iter_2(hashmap: &mut HashMap<usize, usize>) -> IterMut<'_, usize, usize> {
    hashmap.into_iter()
}
#[doc = "pure"]
pub fn into_iter_3(hashmap: &HashMap<usize, usize>) -> Iter<'_, usize, usize> {
    hashmap.into_iter()
}
redefine! { <HashMap<usize, usize>>::into_keys => hashmap: HashMap<usize, usize> => IntoKeys<usize, usize>, "pure" }
redefine! { <HashMap<usize, usize>>::into_values => hashmap: HashMap<usize, usize> => IntoValues<usize, usize>, "pure" }
redefine! { <HashMap<usize, usize>>::is_empty => hashmap: &HashMap<usize, usize> => bool, "pure" }
redefine! { <HashMap<usize, usize>>::iter => hashmap: &HashMap<usize, usize> => Iter<'_, usize, usize>, "pure" }
redefine! { <HashMap<usize, usize>>::iter_mut => hashmap: &mut HashMap<usize, usize> => IterMut<'_, usize, usize>, "impure" }
redefine! { <HashMap<usize, usize>>::keys => hashmap: &HashMap<usize, usize> => Keys<'_, usize, usize>, "pure" }
redefine! { <HashMap<usize, usize>>::len => hashmap: &HashMap<usize, usize> => usize, "pure" }
redefine! { <HashMap<usize, usize>>::ne => hashmap: &HashMap<usize, usize>, other: &HashMap<usize, usize> => bool, "pure" }
redefine! { <HashMap<usize, usize>>::new => => HashMap<usize, usize, RandomState>, "pure" }
redefine! { <HashMap<usize, usize>>::remove => hashmap: &mut HashMap<usize, usize>, k: &usize => Option<usize>, "impure" }
redefine! { <HashMap<usize, usize>>::remove_entry => hashmap: &mut HashMap<usize, usize>, k: &usize => Option<(usize, usize)>, "impure" }
redefine! { <HashMap<usize, usize>>::retain => hashmap: &mut HashMap<usize, usize>, f: impl FnMut(&usize, &mut usize) -> bool => (), "impure" }
redefine! { <HashMap<usize, usize>>::reserve => hashmap: &mut HashMap<usize, usize>, additional: usize => (), "impure" }
redefine! { <HashMap<usize, usize>>::shrink_to => hashmap: &mut HashMap<usize, usize>, min_capacity: usize => (), "impure" }
redefine! { <HashMap<usize, usize>>::shrink_to_fit => hashmap: &mut HashMap<usize, usize> => (), "impure" }
redefine! { <HashMap<usize, usize>>::try_reserve => hashmap: &mut HashMap<usize, usize>, additional: usize => Result<(), TryReserveError>, "impure" }
redefine! { <HashMap<usize, usize>>::values => hashmap: &HashMap<usize, usize> => Values<'_, usize, usize>, "pure" }
redefine! { <HashMap<usize, usize>>::values_mut => hashmap: &mut HashMap<usize, usize> => ValuesMut<'_, usize, usize>, "impure" }
redefine! { <HashMap<usize, usize>>::with_capacity => capacity: usize => HashMap<usize, usize, RandomState>, "pure" }
redefine! { <HashMap<usize, usize>>::with_capacity_and_hasher => capacity: usize, hasher: RandomState => HashMap<usize, usize, RandomState>, "pure" }
redefine! { <HashMap<usize, usize>>::with_hasher => hash_builder: RandomState => HashMap<usize, usize, RandomState>, "pure" }
