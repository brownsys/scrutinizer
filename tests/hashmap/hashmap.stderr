error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:11:1
   |
11 |   redefine! { <HashMap<usize, usize>>::clone_from => hashmap: &mut HashMap<usize, usize>, other: &HashMap<usize, usize> => (), "impure" }
   |   --------------------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: `#[deny(scrutinizer_lint)]` on by default
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:12:1
   |
12 |   redefine! { <HashMap<usize, usize>>::clear => hashmap: &mut HashMap<usize, usize> => (), "impure" }
   |   --------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:14:1
   |
14 |   redefine! { <HashMap<usize, usize>>::drain => hashmap: &mut HashMap<usize, usize> => Drain<'_, usize, usize>, "impure" }
   |   ------------------------------------------------------------------------------------------------------------------------ in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:16:1
   |
16 |   redefine! { <HashMap<usize, usize>>::entry => hashmap: &mut HashMap<usize, usize>, key: usize => Entry<'_, usize, usize>, "impure" }
   |   ------------------------------------------------------------------------------------------------------------------------------------ in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:29:9
   |
29 | /         pub fn $new_ident<$($lt),*>($($param_ident : $param_ty),*) -> $ret_ty {
30 | |             <$origin_ty>::$func_ident($($param_ident),*)
31 | |         }
   | |_________^
   |
  ::: hashmap.rs:17:1
   |
17 |   redefine! { extend_1, <HashMap<usize, usize>>::extend<'a> => hashmap: &mut HashMap<usize, usize>, iter: impl IntoIterator<Item = (&'a usize, &'a usize)> => (), "impure" }
   |   -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:23:9
   |
23 | /         pub fn $new_ident($($param_ident : $param_ty),*) -> $ret_ty {
24 | |             <$origin_ty>::$func_ident($($param_ident),*)
25 | |         }
   | |_________^
   |
  ::: hashmap.rs:18:1
   |
18 |   redefine! { extend_2, <HashMap<usize, usize>>::extend => hashmap: &mut HashMap<usize, usize>, iter: impl IntoIterator<Item = (usize, usize)> => (), "impure" }
   |   -------------------------------------------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:21:1
   |
21 |   redefine! { <HashMap<usize, usize>>::from_iter => iter: impl IntoIterator<Item = (usize, usize)> => HashMap<usize, usize>, "impure" }
   |   ------------------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:22:1
   |
22 |   redefine! { <HashMap<usize, usize>>::fmt => hashmap: &HashMap<usize, usize>, f: &mut Formatter<'_> => FmtResult, "impure" }
   |   --------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:17:9
   |
17 | /         pub fn $func_ident<$($lt),*>($($param_ident : $param_ty),*) -> $ret_ty {
18 | |             <$origin_ty>::$func_ident($($param_ident),*)
19 | |         }
   | |_________^
   |
  ::: hashmap.rs:25:1
   |
25 |   redefine! { <HashMap<usize, usize>>::get_mut<'a> => hashmap: &'a mut HashMap<usize, usize>, k: &usize => Option<&'a mut usize>, "impure" }
   |   ------------------------------------------------------------------------------------------------------------------------------------------ in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:28:1
   |
28 |   redefine! { <HashMap<usize, usize>>::insert => hashmap: &mut HashMap<usize, usize>, k: usize, v: usize => Option<usize>, "impure" }
   |   ----------------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> hashmap.rs:31:1
   |
31 | / pub fn into_iter_2(hashmap: &mut HashMap<usize, usize>) -> IterMut<'_, usize, usize> {
32 | |     hashmap.into_iter()
33 | | }
   | |_^
   |
   = help: consider using sandbox or privacy region

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:42:1
   |
42 |   redefine! { <HashMap<usize, usize>>::iter_mut => hashmap: &mut HashMap<usize, usize> => IterMut<'_, usize, usize>, "impure" }
   |   ----------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:47:1
   |
47 |   redefine! { <HashMap<usize, usize>>::remove => hashmap: &mut HashMap<usize, usize>, k: &usize => Option<usize>, "impure" }
   |   -------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:48:1
   |
48 |   redefine! { <HashMap<usize, usize>>::remove_entry => hashmap: &mut HashMap<usize, usize>, k: &usize => Option<(usize, usize)>, "impure" }
   |   ----------------------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:49:1
   |
49 |   redefine! { <HashMap<usize, usize>>::retain => hashmap: &mut HashMap<usize, usize>, f: impl FnMut(&usize, &mut usize) -> bool => (), "impure" }
   |   ----------------------------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:50:1
   |
50 |   redefine! { <HashMap<usize, usize>>::reserve => hashmap: &mut HashMap<usize, usize>, additional: usize => (), "impure" }
   |   ------------------------------------------------------------------------------------------------------------------------ in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:51:1
   |
51 |   redefine! { <HashMap<usize, usize>>::shrink_to => hashmap: &mut HashMap<usize, usize>, min_capacity: usize => (), "impure" }
   |   ---------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:52:1
   |
52 |   redefine! { <HashMap<usize, usize>>::shrink_to_fit => hashmap: &mut HashMap<usize, usize> => (), "impure" }
   |   ----------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:53:1
   |
53 |   redefine! { <HashMap<usize, usize>>::try_reserve => hashmap: &mut HashMap<usize, usize>, additional: usize => Result<(), TryReserveError>, "impure" }
   |   ----------------------------------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)

error: static analysis was not able to verify the purity of the region
  --> lib.rs:11:9
   |
11 | /         pub fn $func_ident($($param_ident : $param_ty),*) -> $ret_ty {
12 | |             <$origin_ty>::$func_ident($($param_ident),*)
13 | |         }
   | |_________^
   |
  ::: hashmap.rs:55:1
   |
55 |   redefine! { <HashMap<usize, usize>>::values_mut => hashmap: &mut HashMap<usize, usize> => ValuesMut<'_, usize, usize>, "impure" }
   |   --------------------------------------------------------------------------------------------------------------------------------- in this macro invocation
   |
   = help: consider using sandbox or privacy region
   = note: this error originates in the macro `redefine` (in Nightly builds, run with -Z macro-backtrace for more info)
